### DISCUSSION TITLE
UIA in the OneNote UWP

### DISCUSSION THREAD

**Discussion by sovprene** [nvaccess/nvda#17318] [General]
I need to read notes via UIA from the OneNote UWP app (Windows 10 App). I enumerate the elements using IUIAutomationTreeWalker and get to the page element, but there are no childs there. But if I run the application Nvda, then these elements appear. I don’t know python, I didn’t know how  to start debugging and it’s hard for me to understand how this application works. I tried to subscribe to the notification EVENT_SYSTEM_ALERT (as in chrome), but this does not help. Could you tell me what needs to be done to display the  childs items in UWP application?
Someone can help me with this, any idea ?

Thanks You.

**Comment by DrSooom** [comment]
@sovprene: Please completely fill out the bug report draft. Thanks.

**Comment by Brian1Gaff** [comment]
Or maybe this is perhaps a post for the add ons list instead?
 Brian

bglists@blueyonder.co.uk 
Sent via blueyonder. 
Please address personal E-mail to:-
briang1@blueyonder.co.uk, putting 'Brian Gaff'
in the display name field.
-----

**Comment by sovprene** [comment]
This is not a problem with the program, it works correctly. I appeal to the developers, since I have a similar task, but I need to read the text from OneNote with using UIA.

**Comment by Adriani90** [comment]
I think @josephsl might be the best person to answer the question here.

**Comment by josephsl** [comment]
Hi,

The thing is, I'm not really a OneNote user, so can't comment on that. Sorry.

Thanks.

**Comment by Adriani90** [comment]
@josephsl as far as I understand this issue Onenote is used just as an example here. I think the initial author has a general question regarding how to display child objects in UWP apps via UIA.

**Comment by Adriani90** [comment]
Onenote works correctly here, the author wants to implement the same in his or her application also. However, I am not sure if we really can help here. Microsoft would be a better advisor in answering this question in general.
@sovprene did you contact them directly and asked how they did it in Onenote?

**Comment by josephsl** [comment]
Hi, one workaround I did for cases like this is creating a tree interceptor (#8466). Thanks.

**Comment by Adriani90** [comment]
I am converting this to a discussion, in case this is still valid.
